tests/test_ical.py

```diff
@@ -0,0 +1,222 @@
+
+        new_years_day_count = output.count("SUMMARY:New Year's Day\r\n")
+        self.assertEqual(new_years_day_count, 1)
+
+    def test_localized_holiday_names(self):
+        output = self.jp_exporter.generate()
+
+        self.assertIn("SUMMARY:元日\r\n", output)
+
+    def test_empty_holidays(self):
+        exporter = ICalExporter(country_holidays("TH", years=1800))
+        output = exporter.generate()
+
+        self.assertIn("BEGIN:VCALENDAR\r\n", output)
+        self.assertIn("PRODID:", output)
+        self.assertIn("VERSION:2.0\r\n", output)
+        self.assertIn("END:VCALENDAR\r\n", output)
+
+    def test_line_folding_edge_cases(self):
+        line_74_bytes = "x" * 66
+        test_holidays = MockHolidays()._add_custom_holiday(line_74_bytes)
+        self._assert_line_lengths(ICalExporter(test_holidays).generate())
+
+        exact_75_bytes = "x" * 72 + "あ"
+        test_holidays = MockHolidays()._add_custom_holiday(exact_75_bytes)
+        self._assert_line_lengths(ICalExporter(test_holidays).generate())
+
+        needs_adjustment = "x" * 73 + "あ"
+        test_holidays = MockHolidays()._add_custom_holiday(needs_adjustment)
+        self._assert_line_lengths(ICalExporter(test_holidays).generate())
+
+    def test_long_holiday(self):
+        very_long_name = (
+            "This is a very long holiday name that should be folded "
+            "according to RFC 5545 specifications あいうえお"
+        )
+        test_holidays = MockHolidays()._add_custom_holiday(very_long_name)
+        exporter = ICalExporter(test_holidays)
+        output = exporter.generate()
+
+        for line in output.split(CONTENT_LINE_DELIMITER):
+            self.assertLessEqual(len(line.encode()), CONTENT_LINE_MAX_LENGTH)
+
+            if line.startswith(" "):
+                self.assertTrue(line[0].isspace())
+
+    def test_return_bytes(self):
+        self._assert_byte_output(self.us_exporter, b"SUMMARY:New Year's Day")
+        self._assert_byte_output(self.jp_exporter, b"SUMMARY:\xe5\x85\x83\xe6\x97\xa5")
+
+    def test_crlf_line_endings(self):
+        output = self.th_exporter.generate()
+
+        lines = output.splitlines(True)
+        for line in lines:
+            self.assertTrue(
+                line.endswith(CONTENT_LINE_DELIMITER), f"Line did not end with CRLF: {repr(line)}"
+            )
+
+    def test_unique_uid_generation(self):
+        output = self.us_exporter.generate()
+        uids = [line for line in output.split(CONTENT_LINE_DELIMITER) if line.startswith("UID:")]
+
+        self.assertEqual(len(uids), len(set(uids)), "Duplicate UIDs found in iCal output.")
+
+    def test_empty_summary_handling(self):
+        test_holidays = MockHolidays()._add_custom_holiday("")
+        exporter = ICalExporter(test_holidays)
+
+        with self.assertRaises(ValueError) as context:
+            exporter.generate()
+
+        self.assertEqual(str(context.exception), "Holiday name cannot be empty.")
```

Comment:
This test causes an infinite loop.

Reply:
Holidays does not fully support holidays with an empty name.

Reply:
Fixed in https://github.com/vacanza/holidays/pull/2362/commits/800a547f4fbd9d7faf7c9dc1b2e6709944a1cabf